src/dsp_tools/utils/shared.py

```diff
@@ -75,7 +75,7 @@ def try_network_action(action: Callable[..., Any]) -> Any:
             time.sleep(2 ** i)
             continue
         except BaseError as err:
-            if regex.search(r"try again later", err.message) or regex.search(r"status code=5\d\d", err.message):
+            if regex.search(r"try again later", err.message) or regex.search(r"5\d\d", str(err.status_code_of_api_response)):
```

Comment:
perhaps not the most elegant way. But if I wanted to analyze `err.status_code_of_api_response` as integer, 
 - I first have to check if it is not None, 
 - then check that it's > 500
 - then check that it's < 600